cassandra_test.go

```diff
@@ -3288,3 +3288,46 @@ func TestQuery_NamedValues(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestBatchKeyspaceField(t *testing.T) {
+	session := createSession(t)
+	defer session.Close()
+
+	if session.cfg.ProtoVersion < protoVersion5 {
+		t.Skip("keyspace for BATCH message is not supported in protocol < 5")
+	}
+
+	err := createTable(session, "CREATE TABLE batch_keyspace(id int, value text, PRIMARY KEY (id))")
```

Comment:
You should create table in different keyspace, and then do `b.keyspace = "foo"`. Query `session.Query("SELECT * FROM batch_keyspace")` should also override the default keyspace.

Reply:
This has been addressed in #1822.